wordpress_connector/models/match_partner_wp.py

```diff
@@ -79,5 +79,7 @@ def _match_get_rules_order(self):
     @api.model
     def _match_rule_child_id(self, partner_obj, infos, options=None):
         # if a keyerror is raise it is handled as "no child found go to next rule"
-        child_id = infos.pop("child_id")
-        return self.env["compassion.child"].search([("local_id", "like", child_id)]).sponsor_id
+        child_local_id = infos.pop("child_id")
+        child = self.env["compassion.child"].search([("local_id", "like", child_local_id)])
+        sponsorship = self.env["recurring.contract"].search([("child_id", "=", child.id)])
```

Comment:
This would ensure to only take the latest sponsorship.
```suggestion
        sponsorship = self.env["recurring.contract"].search([("child_id", "=", child.id)], limit=1)
```